pyne/vta.py

```diff
@@ -0,0 +1,551 @@
+    tp : Point
+        Temporary point position.
+    end : Point
+        End point
+    v : list
+        Direction vector from start to end: [v_x, v_y, v_z].
+    bounds : list
+        Boundaries of the mesh grid.
+
+    Return:
+    -------
+    t_maxs : list
+        The maximum travel lenght in current voxel for 3 directions. Format: 
+        [t_max_x, t_max_y, t_max_z]. These value could be both positive and
+        negtive. And could be 'inf' or '-inf'.
+    """
+    n_x_grid = _find_next_grid_1d(tp.x, v[0], bounds[0])
+    n_y_grid = _find_next_grid_1d(tp.y, v[1], bounds[1])
+    if v[0] == 0.0:
+        t_max_x = float('inf')
+    else:
+        t_max_x = (n_x_grid - tp.x) / v[0]
+    if v[1] == 0.0:
+        t_max_y = float('inf')
+    else:
+        t_max_y = (n_y_grid - tp.y) / v[1]
+    return [t_max_x, t_max_y]
+
+def _move_to_next_voxel(x_idx, y_idx, z_idx, tp, t_temp, t_maxs, v):
+    """
+    Move the point to next voxel.
+
+    Parameters:
+    -----------
+    x_idx : int
+        Mesh idx in x direction.
+    y_idx : int
+        Mesh idx in y direction.
+    z_idx : int
+        Mesh idx in z direction.
+    tp : Point
+        Temporary point position.
+    t_temp: float
+        Temporary travel lenght from the start.
+    t_maxs: list
+        The maximum travel length in current voxel for 3 directions.
+    v : list
+        Direction vector from start to end.
+
+    Returns:
+    --------
+    x_idx : int
+        Modified x_idx.
+    y_idx : int
+        Modified y_idx.
+    z_idx : int
+        Modified z_idx.
+    tp : Point
+        Point after moving along the direction of v.
+    t_temp : float
+        Modifed travel length from the start.
+    """
+    t_min = min(t_maxs)
+    if t_maxs[0] == t_min:
+        x_idx += 1
+    elif t_maxs[1] == t_min:
+        y_idx += 1
+    elif t_maxs[2] == t_min:
+        z_idx += 1
+    tp.x += t_min * v[0]
+    tp.y += t_min * v[1]
+    tp.z += t_min * v[2]
+    t_temp += t_min
+    return x_idx, y_idx, z_idx, tp, t_temp
+
+def _move_to_boundary(tp, t_min, v):
+    """
+    Move the point to the nearest boundary when Initialize the problem.
+
+    Parameters:
+    -----------
+    tp : Point
+        Temporary point. Actually, it is also the start point.
+    t_min : float
+        Distance to the nearest (positive) boundary. For a 3D mesh, there are 6
+        outside boundaries. Therefore there are 6 value of the distance from the
+        start to the boundaries. This t_min is the smallest one among the
+        positive distance. (The distance is negtive when the direction is
+        negtive.)
+    v : list
+        Direction vector.
+
+    Return:
+    -------
+    tp : Point
+        Modified point. This point lies on the nearest boundary. Therefore, we
+        moved point from outside the mesh to the mesh boundary (or the extension
+        of the boundary, which still outside the mesh).
+    """
+
+    # calculate the new coordinate
+    tp.x += t_min * v[0]
```

Comment:
This only updates two dimensions?

Reply:
Sorry again. Another place I forgot to update. It should update three dimensions.